external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -170,10 +182,13 @@ private void initialize(Collection<TopicPartition> partitions) {
                 final long fetchOffset = doSeek(tp, committedOffset);
                 setAcked(tp, fetchOffset);
             }
-            initialized = true;
+
+            task.start();
```

Comment:
As far as I recall the KafkaConsumer won't let you commit messages to partitions the consumer is not assigned, so committing after onPartitionsRevoked won't work. I think it would be a nicer solution to eliminate most rebalances by switching to manual partition assignment